radix-engine-interface/src/api/component/substates.rs

```diff
@@ -21,5 +21,5 @@ pub struct ComponentRoyaltyConfigSubstate {
 
 #[derive(Debug, Clone, PartialEq, Eq, ScryptoSbor)]
 pub struct ComponentRoyaltyAccumulatorSubstate {
-    pub royalty_vault: Own,
+    pub royalty_vault: Option<Own>,
```

Comment:
Mainly because of the chicken-and-egg problem.  If every object, including package objects, has royalty which requires vault creation during creation time, there is no way to publish package.

As we discussed earlier, if we flush package subsates directly in bootstrap, I will be more than happy to make it non-optional.